### PR TITLE
nix-info: fix error when no channel installed

### PR DESCRIPTION
**Pull request by peng1999**
###### Description of changes

Modify the nix-info script so that it doesn't fails when user only use flakes and never installed any channels. This could happen when user installed nix using non-standard ways, e.g. [using archlinux's package manager](https://archlinux.org/packages/community/x86_64/nix/).

###### Things done

This patch use the new nix CLI when old fails.

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peng1999** [comment]
@1000101 @zowoq 
Could you please review this PR?

File: pkgs/tools/nix/info/info.sh [hunk 1]
```diff
@@ -69,8 +69,14 @@ nixev() {
...
 
+# use `nix eval` command
+nixev2() {
+    nix eval --raw "$1"
+}
+
 desc_system() {
-    nixev '(import <nixpkgs> {}).stdenv.hostPlatform.system'
+    nixev '(import <nixpkgs> {}).stdenv.hostPlatform.system' 2>/dev/null ||
+        nixev2 'nixpkgs#stdenv.hostPlatform.system'
```
Comment (NickCao) [inline pkgs/tools/nix/info/info.sh:79]: Entry for nixpkgs is not guaranteed to be in the flake registry either. (Some overrides the registry to custom values or disables the registry completely)

File: pkgs/tools/nix/info/info.sh [hunk 2]
```diff
@@ -69,8 +69,14 @@ nixev() {
     nix-instantiate --eval --strict -E "$1"
 }
 
+# use `nix eval` command
+nixev2() {
+    nix eval --raw "$1"
```
Comment (NickCao) [inline pkgs/tools/nix/info/info.sh:74]: The nix3 cli is only available in nix 2.4+, and is guarded by experimental flags. There should be a version check and the relavent flags should be set.

**Review by NickCao — COMMENTED** [review]
Overall I think the implementation can still not cover the edge cases. Can't we just handle the errors gracefully and simply print unknown for flake only systems.

File: pkgs/tools/nix/info/info.sh (hunk 1 above)
Comment (figsoda) [reply pkgs/tools/nix/info/info.sh:79]: What does the nixpkgs here do exactly, isn't this just `builtins.currentSystem`?

Comment (peng1999) [reply pkgs/tools/nix/info/info.sh]: Oh yes,`builtins.currentSystem` seems to do exactly what we want and works everywhere. I have no idea why originally nixpkgs is used. Changed.

File: pkgs/tools/nix/info/info.sh (hunk 2 above)
Comment (peng1999) [reply pkgs/tools/nix/info/info.sh]: Change has made according to @figsoda 's comment. There is no need to use experimental command now.

**Review by NickCao — APPROVED** [review]

**Pull request merged**